### PR TITLE
Dump/Restore Docs

### PR DESCRIPTION
**Pull request by ravenac95**
TODO:

* [x] Add upgrade parameters section to the current testnet parameters

### CODE REVIEW

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+::: tip NOTE
+Do not stop your `oasis-node` process just yet.
+:::
+
+Before an upgrade we will update the [Upgrade
+Parameters](./current-testnet-parameters.md#upgrade-parameters) to contain the
+block height to dump.
+
+```bash
+export HEIGHT_TO_DUMP=<height_to_dump>
```
Comment (ptrus) [inline src/operators/maintenance/upgrading-your-nodes.md]: this needs to be done after the node has reached that state right? should probably note that

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+export PATCH_URL=<url_to_patch>
+curl -o /serverdir/etc/patch.json $PATCH_URL
+```
+
+### Apply the Patch
+
+```bash
+jq --slurp --sort-keys --compact-output '.[0] * .[1]' \
+  /serverdir/etc/upgrade-dump.json \
+  /serverdir/etc/patch.json > /serverdir/etc/genesis.json
```
Comment (ptrus) [inline src/operators/maintenance/upgrading-your-nodes.md]: I think that at least for first few upgrades we should probably also provide our exported genesis so node operators can confirm they did the process corectly

Comment (ravenac95) [reply src/operators/maintenance/upgrading-your-nodes.md]: Oh ya, we will also do that :). I just wanted to provide some instructions for doing it on your own, but based on the commands I used it should be possible to get a reliable shasum (i tested).

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Upgrading your nodes
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
# Network Upgrades
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+```bash
+export HEIGHT_TO_DUMP=<height_to_dump>
+oasis-node genesis dump \
+  -a unix:/serverdir/node/internal.sock \
+  --genesis.file /serverdir/etc/upgrade-dump.json \
+  --height $HEIGHT_TO_DUMP
+```
+
+## Stop your node
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
## Stopping Your Node
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+We will provide a state patch [here](../current-testnet-parameters.md) that you
+can apply using the `jq` tool. We suggest that you install this on your server
+so that you can do all the upgrade process without having to transfer anything
+onto your local machine. The patched state will update at least the following
+fields:
+
+* `
+
+### Download the Patch
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
### Downloading the Patch
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+```bash
+export PATCH_URL=<url_to_patch>
+curl -o /serverdir/etc/patch.json $PATCH_URL
+```
+
+### Apply the Patch
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
### Applying the Patch
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+::: warning WARNING
+We do not suggest that you wipe _all_ state. You might lose node identities and
+keys if you do it this way.
+:::
+
+Before restarting your node you should wipe tendermint state. The process for
+this is described [here](./wiping-node-state.md#state-wipe-and-keep-node-identity).
+
+## Restarting the node
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
## Restarting the Node
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+This will depend on your process manager. If you don't have a process manager
+you should use one. However, to start the node without an office manager you can
+start the `oasis-node` like so:
+
+```bash
+oasis-node --config /serverdir/etc/config.yml
+```
+
+## Cleaning up
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
## Cleaning Up
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+keys if you do it this way.
+:::
+
+Before restarting your node you should wipe tendermint state. The process for
+this is described [here](./wiping-node-state.md#state-wipe-and-keep-node-identity).
+
+## Restarting the node
+
+This will depend on your process manager. If you don't have a process manager
+you should use one. However, to start the node without an office manager you can
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
you should use one. However, to start the node without a process manager you can
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
+# Upgrading your nodes
+
+::: warning WARNING
+Following this guide when there is no software upgrade would result in you
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
Following this guide when there is no network upgrade would result in you
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+losing your place on the current network.
+:::
+
+The following guide should be used when the network has agreed to do a software
+upgrade for the network.
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
upgrade.
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+## Dumping Blockchain State
+
+::: tip NOTE
+Do not stop your `oasis-node` process just yet.
+:::
+
+Before an upgrade we will update the [Upgrade
+Parameters](./current-testnet-parameters.md#upgrade-parameters) to contain the
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
Parameters](./current-testnet-parameters.md#upgrade-parameters) to specify the
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+block height to dump.
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]:
```suggestion
block height at which to dump.
```

File: src/operators/maintenance/upgrading-your-nodes.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+## Patching Dumped State
+
+We will provide a state patch [here](../current-testnet-parameters.md) that you
+can apply using the `jq` tool. We suggest that you install this on your server
+so that you can do all the upgrade process without having to transfer anything
+onto your local machine. The patched state will update at least the following
+fields:
+
+* `
```
Comment (peterjgilbert) [inline src/operators/maintenance/upgrading-your-nodes.md]: TODO: list these fields

File: src/operators/maintenance/upgrading-your-nodes.md (hunk 3 above)
Comment (peterjgilbert) [reply src/operators/maintenance/upgrading-your-nodes.md]: Would suggest renaming this file to `handling-network-upgrades.md` as well

File: src/operators/current-testnet-parameters.md [hunk 15]
```diff
@@ -4,30 +4,57 @@ This page is meant to be kept up to date with the information from the currently
...
+[Handling Network Upgrades Guide](./maintenance/handling-network-upgrades.md)
+for instructions on how to use these parameters.
+
+* Block height to dump: **950000**
+* Upgrade Window
+  * Start: **2019-12-17T19:00:00Z**
+  * End: **2019-12-18T18:59:59Z**
+    * The "_End_" of the window is not something we can enforce unilaterally,
+      however, if, for whatever reason, not enough people upgrade on the public
+      testnet we _may_ release and redeploy a new genesis block that removes
```
Comment (peterjgilbert) [inline src/operators/current-testnet-parameters.md]: This is incomplete

File: src/operators/current-testnet-parameters.md [hunk 16]
```diff
@@ -4,30 +4,57 @@ This page is meant to be kept up to date with the information from the currently
...
 
-* [Genesis Document](https://github.com/oasislabs/public-testnet-artifacts/releases/download/2019-11-26/genesis.json):
-  * sha1: `bed77f6433e607f7073a48dbf27f8cf5d9a8c2e1`
-  * sha256: `ee38b53a2e8acd785d279e4c8d3af38b832f120d9a85bf0173e94878f45bc718`
+::: tip NOTE
+If you're upgrading your node you should use the [Upgrade
+Parameters](#upgrade-parameters) section.
+:::
+
+* Genesis Document _No link at this time_:
```
Comment (peterjgilbert) [inline src/operators/current-testnet-parameters.md]: Could add an "expected release" time here.

**Pull request merged**